# CÂU 2: PHÂN TÍCH TỶ SUẤT LỢI NHUẬN GỘP THEO PHÂN KHÚC

**1. Yêu cầu bài toán:**
Tìm phân khúc sản phẩm (**segment**) có tỷ suất lợi nhuận gộp trung bình cao nhất trong file `products.csv`.
* Công thức tính tỷ suất lợi nhuận gộp: `(price - cogs) / price`

**2. Phương pháp tiếp cận & Suy luận:**
Để xác định phân khúc sinh lời tốt nhất, chúng ta thực hiện quy trình sau:
* **Bước 1:** Tải dữ liệu từ file `products.csv`.
* **Bước 2:** Tạo cột dữ liệu mới tính toán tỷ suất lợi nhuận cho từng sản phẩm cụ thể dựa trên công thức đề bài cung cấp.
* **Bước 3:** Sử dụng phương pháp gom nhóm (`groupby`) theo từng phân khúc (Premium, Performance, Activewear, Standard).
* **Bước 4:** Tính giá trị trung bình (`mean`) của tỷ suất lợi nhuận cho mỗi nhóm và sắp xếp kết quả để tìm ra nhóm cao nhất.

**3. Lý do chọn phương pháp:**
* **Tính toán trực tiếp trên mảng (Vectorization):** Phép tính `(price - cogs)/price` được áp dụng trực tiếp lên toàn bộ mảng dữ liệu (Series) của Pandas, tận dụng lõi xử lý C dưới nền tảng giúp tăng tốc độ thực thi lên mức tối đa.
* **Tại sao dùng `groupby` và `mean`?** Bài toán yêu cầu đánh giá cả một "phân khúc" (segment). Do đó, hàm `groupby` là công cụ chuẩn xác nhất để phân lập các nhóm. Phép tính trung bình cộng (`mean()`) được chọn thay vì tính tổng (`sum()`) vì nó phản ánh đúng tỷ suất sinh lời đại diện của một sản phẩm bất kỳ trong nhóm đó, giúp ban quản trị có cái nhìn công bằng giữa nhóm bán được số lượng nhiều và nhóm bán được số lượng ít.

In [ ]:
# Import thư viện
import pandas as pd
from google.colab import drive

# Kết nối với Google Drive
drive.mount('/content/drive')

Mounted at /content/drive


**Thực hiện Bước 1 & Bước 2: Tải dữ liệu và Tính toán tỷ suất lợi nhuận**

Chúng ta nạp dữ liệu sản phẩm và áp dụng công thức vector hóa của Pandas để tính toán nhanh chóng cho toàn bộ danh mục.

In [ ]:
# 1. Khai báo đường dẫn
file_path = '/content/drive/MyDrive/datathon-2026-round-1/products.csv'

# 2. Đọc dữ liệu
df_products = pd.read_csv(file_path)

# 3. Tính tỷ suất lợi nhuận gộp (Profit Margin) cho từng sản phẩm
# Công thức: (Giá bán - Giá vốn) / Giá bán
df_products['profit_margin'] = (df_products['price'] - df_products['cogs']) / df_products['price']

# Hiển thị 5 dòng đầu để kiểm tra cột mới
df_products[['product_name', 'segment', 'price', 'cogs', 'profit_margin']].head()

,product_name,segment,price,cogs,profit_margin
0,SaigonFlex UC-01,Everyday,11059.650000,9704.842875,0.1225
1,SaigonFlex UC-02,Everyday,9523.076013,5393.870254,0.4336
2,SaigonFlex UC-03,Everyday,15951.633158,11371.919278,0.2871
3,SaigonFlex UC-04,Everyday,15753.717299,8573.172954,0.4558
4,SaigonFlex UC-05,Everyday,15766.334536,14063.570406,0.1080


**Thực hiện Bước 3 & Bước 4: Gom nhóm và So sánh kết quả**

Chúng ta tính trung bình cộng của tỷ suất lợi nhuận cho từng phân khúc để đưa ra cái nhìn tổng quát về hiệu quả kinh doanh của từng nhóm sản phẩm.

In [ ]:
# Gom nhóm theo segment và tính trung bình profit_margin
segment_performance = df_products.groupby('segment')['profit_margin'].mean().sort_values(ascending=False)

print("Tỷ suất lợi nhuận gộp trung bình theo từng phân khúc:")
print(segment_performance)

# Xác định phân khúc cao nhất
best_segment = segment_performance.idxmax()
max_value = segment_performance.max()

print(f"\n--> Phân khúc có tỷ suất lợi nhuận gộp trung bình cao nhất là: {best_segment} ({max_value:.4f})")

Tỷ suất lợi nhuận gộp trung bình theo từng phân khúc:
segment
Standard       0.313442
Premium        0.285377
All-weather    0.284176
Activewear     0.265600
Performance    0.263650
Balanced       0.258038
Trendy         0.240758
Everyday       0.236343
Name: profit_margin, dtype: float64

--> Phân khúc có tỷ suất lợi nhuận gộp trung bình cao nhất là: Standard (0.3134)


**KẾT LUẬN CUỐI CÙNG:**
Dựa trên kết quả tính toán từ tập dữ liệu `products.csv`, phân khúc có tỷ suất lợi nhuận gộp trung bình cao nhất là **Standard** (với giá trị khoảng 0.3134 hay 31.34%).

Đối chiếu với các phương án của đề bài:

A) Premium

B) Performance

C) Activewear

D) Standard

**=> Đáp án được chọn là: D**